### ISSUE TITLE
unicode-width should retain

### ISSUE THREAD

**Issue by Manishearth** [unicode-rs/unicode-width#64]
As described in https://github.com/unicode-rs/unicode-width/issues/55, 0.1.13 changed the predicted widths of a lot of characters to be more accurate for things like emoji.

From that issue, and from the breakage in rustfmt, it feels useful to have an API that is _explicitly_ stable, and documented as such. Probably one that did the original less-useful-but-"still works" thing of relying on East Asian Width properties only.

**Comment by Manishearth** [comment]
@Jules-Bertholet how hard would it be for you to add a `width_stable`?

**Comment by Manishearth** [comment]
I think the easy way to do this with the traits would be to add a `UnicodeStableWidth` trait. We can clean this up in a future breaking release.

It's actually unclear to me why we have UnicodeWidthStr and UnicodeWidthChar.

**Comment by Jules-Bertholet** [comment]
For *perfect* stability, we'd have to pin a Unicode version, as the underlying Unicode properties are not stable (even for assigned codepoints).

**Comment by Manishearth** [comment]
I'm fine with it being just EAW and only subject to change with Unicode EAW changes, which change _extremely_ rarely.

**Comment by Jules-Bertholet** [comment]
We'd probably want EAW + `Default_Ignorable_Code_Point` + `Grapheme_Extend` + Hangul jungseong/jongseong (which is more or less what earlier versions of this crate did). It might be better to release it as a separate crate though?

**Comment by Manishearth** [comment]
Yeah, sorry, I mean "what we did before", especially since rustfmt relies on this crate.

I think given where this crate is in the ecosystem I'd prefer for this option to be readily available.
